### PR TITLE
Windows: Fix access log tests

### PR DESCRIPTION
**Pull request by sunjayBhatia**
Commit Message: Windows: Fix access log tests
Additional Description:
- formatter test:
  - Anchor the command regex at the beginning of the search space to replace
    the existing invalid use of the match position logic (match should not
    be used at all when the regex search fails).
  - add test case that demonstrates anchor is needed, removal of anchor
    causes new test case to fail
- benchmark test:
  - benchmark main now matches test/main.cc to initialize the socket stack
    on Windows and to disable the CRT invalid parameter abort behavior
- also refactors all test main.cc files to use the common TestEnvironment::initializeTestMain 
  method to reduce duplication

Risk Level: Low
Testing: Adds additional access log formatter unit test case 
Docs Changes: N/A
Release Notes: N/A

### CODE REVIEW

**Comment by sunjayBhatia** [comment]
cc @wrowe

File: test/benchmark/main.cc [hunk 1]
```diff
@@ -1,16 +1,36 @@
...
 int main(int argc, char** argv) {
+#if defined(WIN32)
+  _set_abort_behavior(0, _WRITE_ABORT_MSG | _CALL_REPORTFAULT);
+
+  _set_invalid_parameter_handler(NoopInvalidParameterHandler);
+
+  WSADATA wsa_data;
+  const WORD version_requested = MAKEWORD(2, 2);
+  RELEASE_ASSERT(WSAStartup(version_requested, &wsa_data) == 0, "");
+#endif
```
Comment (mattklein123) [inline test/benchmark/main.cc:32]: This is repeated somewhere else. Can you have a common win32 utility function that you call from the mains?

Comment (sunjayBhatia) [reply test/benchmark/main.cc:32]: Seems like this block is also repeated a lot: https://github.com/envoyproxy/envoy/blob/11eecae656d868be92540e0276e60fc2e80efe88/test/fuzz/main.cc#L57-L63

We could pull all of this out into a "test main init" somewhere in the test utils if that makes sense?

Comment (mattklein123) [reply test/benchmark/main.cc:32]: Sure any cleanup appreciated.

**Comment by mattklein123** [comment]
Apologies can you pay it forward and fix clang tidy? Thank you.

/wait

**Comment by mattklein123** [comment]
@sunjayBhatia please *never* force push as it makes reviews very difficult. Thank you!

**Comment by sunjayBhatia** [comment]
@mattklein123 ah sorry, just did it to fix the signoff on the latest commit to fix clang-tidy

**Pull request merged**